ipa-core/src/protocol/dp/insecure.rs

```diff
@@ -238,4 +355,72 @@ mod test {
             assert!(f64::abs(sample_variance - dp.rounded_normal_dist.std().powi(2)) < 2.0);
         }
     }
+
+    /// Tests for OPRF Padding DP
+    #[test]
+    fn test_pow_u32() {
+        assert!(is_close(pow_u32(2.0, 4), 16.0, 5));
+        assert!(is_close(pow_u32(6.0, 3), 216.0, 5));
+        assert!(is_close(pow_u32(0.0, 0), 1.0, 5));
+    }
+
+    #[test]
+    fn test_find_smallest_n() {
+        assert_eq!(find_smallest_n(1, 0.5, 1e-6), 25);
+        assert_eq!(find_smallest_n(1, 1.0, 1e-06), 14);
+        assert_eq!(find_smallest_n(1, 0.1, 1e-06), 109);
+        assert_eq!(find_smallest_n(1, 0.01, 1e-06), 852);
+        assert_eq!(find_smallest_n(1, 1.0, 1e-07), 16);
+        assert_eq!(find_smallest_n(1, 0.1, 1e-07), 132);
+        assert_eq!(find_smallest_n(1, 0.01, 1e-07), 1082);
+        assert_eq!(find_smallest_n(1, 1.0, 1e-08), 18);
+        assert_eq!(find_smallest_n(1, 0.1, 1e-08), 155);
+        assert_eq!(find_smallest_n(1, 0.01, 1e-08), 1313);
+        assert_eq!(find_smallest_n(10, 1.0, 1e-06), 23);
+        assert_eq!(find_smallest_n(10, 0.1, 1e-06), 137);
+        assert_eq!(find_smallest_n(10, 0.01, 1e-06), 1087);
+        assert_eq!(find_smallest_n(10, 1.0, 1e-07), 25);
+        assert_eq!(find_smallest_n(10, 0.1, 1e-07), 160);
+        assert_eq!(find_smallest_n(10, 0.01, 1e-07), 1317);
+        assert_eq!(find_smallest_n(10, 1.0, 1e-08), 28);
+        assert_eq!(find_smallest_n(10, 0.1, 1e-08), 183);
+        assert_eq!(find_smallest_n(10, 0.01, 1e-08), 1548);
+        assert_eq!(find_smallest_n(100, 1.0, 1e-06), 113);
+        assert_eq!(find_smallest_n(100, 0.1, 1e-06), 231);
+        assert_eq!(find_smallest_n(100, 0.01, 1e-06), 1366);
+        assert_eq!(find_smallest_n(100, 1.0, 1e-07), 115);
+        assert_eq!(find_smallest_n(100, 0.1, 1e-07), 254);
+        assert_eq!(find_smallest_n(100, 0.01, 1e-07), 1597);
+        assert_eq!(find_smallest_n(100, 1.0, 1e-08), 118);
+        assert_eq!(find_smallest_n(100, 0.1, 1e-08), 277);
+        assert_eq!(find_smallest_n(100, 0.01, 1e-08), 1827);
+        assert_eq!(find_smallest_n(1000, 1.0, 1e-06), 1013);
+        assert_eq!(find_smallest_n(1000, 0.1, 1e-06), 1131);
+        assert_eq!(find_smallest_n(1000, 0.01, 1e-06), 2312);
+        assert_eq!(find_smallest_n(1000, 1.0, 1e-07), 1015);
+        assert_eq!(find_smallest_n(1000, 0.1, 1e-07), 1154);
+        assert_eq!(find_smallest_n(1000, 0.01, 1e-07), 2542);
+        assert_eq!(find_smallest_n(1000, 1.0, 1e-08), 1018);
+        assert_eq!(find_smallest_n(1000, 0.1, 1e-08), 1177);
+        assert_eq!(find_smallest_n(1000, 0.01, 1e-08), 2773);
+    }
+    #[test]
+    fn test_oprf_padding_dp() {
+        let oprf_padding = OPRFPaddingDp::new(1.0, 1e-6, 10);
+
+        let mut rng = rand::thread_rng();
+
+        oprf_padding.unwrap().sample(&mut rng);
+    }
+    fn test_oprf_padding_dp_constructor() {
```

Comment:
Did you intend to do this?

```suggestion
    #[test]
    fn test_oprf_padding_dp_constructor() {
```